erts/emulator/drivers/win32/registry_drv.c

```diff
@@ -318,8 +318,12 @@ reg_from_erlang(ErlDrvData clientData, char* buf, ErlDrvSizeT count)
 	}
 	break;
     case CMD_DELETE_KEY:
-	result = RegDeleteKey(rp->hkey, NULL);
-	reply(rp, result);
+	{
+	    char* key = buf;
+
+	    result = RegDeleteKey(rp->hkey, key);
+	    reply(rp, result);
+	}
```

Comment:
I don't get how this could work (other than pure luck). The data buffer sent for CMD_DELETE_KEY does only contain one byte (the value 7). The pointer `buf` has been incremented (before the `switch`) to point just beyond that first and only byte into unknown territory. Passing it as argument to any function is just wrong.

According to `RegDeleteKey` documentation, the second argument `lpSubKey` should be the name of the subkey to be deleted.

Reply:
Maybe better (as @b-winkler proposed) to replace the broken `delete_key` with a new `delete_subkey` which is simpler to implement and does not have the quirk to invalidate the current key.

Reply:
This necessitates a change in the documentation as well, which is why I avoided it initially.  If we're certain we prefer it, I can make that change.  Let me know :)

Reply:
Well, if you have done a `delete_key` that actually works I'm fine with keeping it.